task/sast-unicode-check/0.1/sast-unicode-check.yaml

```diff
@@ -0,0 +1,281 @@
+                SCAN_PROP="find-unicode-control-git-url:${FIND_UNICODE_CONTROL_GIT_URL}#${git_url_suffix}"
+            fi
+
+            # Check usage of find-unicode-control to confirm the tool can work
+            USAGE_EXIT_CODE=0
+            ./find-unicode-control/find_unicode_control.py --help >&2 || USAGE_EXIT_CODE=$?
+            if [[ "${USAGE_EXIT_CODE}" -eq 0 ]]; then
+                echo "Message: Succeed to check usage of find-unicode-control" >&2
+            else
+                echo "Error: Failed to check usage of find-unicode-control" >&2
+                return 1
+            fi
+        }
+
+        # Find unicode control
+        unicontrol_scan()
+        {
+            FUC_EXIT_CODE=0
+            LANG=en_US.utf8 ./find-unicode-control/find_unicode_control.py ${FIND_UNICODE_CONTROL_ARGS} "${SOURCE_CODE_DIR}/source" \
+                >raw_sast_unicode_check_out.txt \
+                2>raw_sast_unicode_check_out.log \
+                || FUC_EXIT_CODE=$?
+
+            if [[ "${FUC_EXIT_CODE}" -eq 0 || "${FUC_EXIT_CODE}" -eq 1 ]]; then
+                # No non-printable unicode characters found, or successfully found
+                echo "Message: Succeed to run find-unicode-control command" >&2
+            else
+                # Failed to run the command
+                echo "Error: failed to run find unicode control command" >&2
+                return 1
+            fi
+
+            # Translate the output format
+            TRANSLATE_EXIT_CODE=0
+            sed -i raw_sast_unicode_check_out.txt -E -e 's|(.*:[0-9]+)(.*)|\1: warning:\2|' -e 's|^|Error: UNICONTROL_WARNING:\n|' || TRANSLATE_EXIT_CODE=$?
+            if [[ "${TRANSLATE_EXIT_CODE}" -eq 0 ]]; then
+                echo "Message: Succeed to translate the unicontrol output format" >&2
+            else
+                echo "Error: failed to translate the unicontrol output format" >&2
+                return 1
+            fi
+        }
+
+        # Process all results as configured with CSGERP_OPTS
+        process_all_results()
+        {
+            CSGERP_OPTS=(
+                --mode=json
+                --remove-duplicates
+                --embed-context=3
+                --set-scan-prop="${SCAN_PROP}"
+                --strip-path-prefix="${SOURCE_CODE_DIR}"/source/
+            )
+            # In order to generate csdiff/v1, we need to add the whole path of the source code as sast-unicode-check only provides an URI to embed the context
+            csgrep "${CSGERP_OPTS[@]}" raw_sast_unicode_check_out.txt > processed_sast_unicode_check_out.json
+
+            csgrep --mode=evtstat processed_sast_unicode_check_out.json
+        }
+
+        # Filter known false positive
+        csfilter_kfp()
+        {
+            echo "Message: Filtering false positives in results files using ${KFP_GIT_URL}..." >&2
+
+            # Build initial csfilter-kfp command
+            csfilter_kfp_cmd="csfilter-kfp --verbose --kfp-git-url=${KFP_GIT_URL}"
```

Comment:
I think the reference to the image should change as `csfilter-kfp` the tool is not present

Reply:
Updated, add some code when `csfilter-kfp` run failed